ext_py/tests/test_matrix.py

```
@@ -0,0 +1,386 @@
+    assert [1, 0, 2, 1, 1] in rows
+    assert [0, 1, 1, 0, 1] in rows
+    # The recorded image rows are members of the image subspace.
+    for r in img.basis:
+        assert img.contains(r)
+
+
+
+@pytest.mark.xfail(reason="Raises PanicException")
+def test_compute_kernel_requires_row_reduce():
+    m = fp.AugmentedMatrix2.from_py(3, [[1, 0, 1], [0, 1, 1]])
+    with pytest.raises(ValueError):
```

Ruff (0.15.20)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)